index.js

```diff
@@ -221,7 +221,11 @@ function parse (uri, opts) {
 
     // fix port number
     if (isNaN(parsed.port)) {
-      parsed.port = matches[5]
+      if (matches[5] === undefined) {
+        parsed.port = parsed.scheme === 'http' ? 80 : 443
```

Comment:
This assumes that we only parse `http`/`https` uri schemes. We support more than this.
```suggestion
        switch (parsed.scheme) {
        case 'http':
        parsed.port = 80
        case 'https':
        parsed.port = 443
        }
```

Reply:
This is not Go. I'd have to add in `break` statements. In my opinion, this is the exact scenario for a ternary.

Reply:
Ternary assumes that we resolves only `http` or `https`; which is not true, we support any scheme; like `ftp` `sftp` etc.

At the same time i don't feel this is a good thing because we're assuming the port based on the scheme; but the uri doesn't provide any.

Reply:
If this library is going to assert that it fully parses URIs, it must parse the port portion:

https://datatracker.ietf.org/doc/html/rfc3986#section-3.2
> authority   = [ userinfo "@" ] host [ ":" port ]

>   URI producers and normalizers should omit the ":" delimiter that
>   separates host from port if the port component is empty.  Some
>   schemes do not allow the userinfo and/or port subcomponents.

In other words, `http://example.com/` translates to:

+ `http`
+ `example.com`
+ `80`

Reply:
That's not true. In `http://example.com` there's no port from the `uri` point of view. Automatically translating to `80` for `http` is an extension of the interpretation of the rfc, not what the rfc dictactes.